### PR TITLE
Export libsession constants

### PR DESCRIPTION
**Pull request by yougotwill**
- added constants wrapper to export user_profile, contact, group and community max name length variables
- user info now truncate names if they are too long
- `setUserInfo` will now return the final name result (in case of truncation)
- added `yarn clean`
- rename `Readme.md` to `README.md`
- include node-api-headers in cmake build process which fixes intellisense issues

### CODE REVIEW

File: shared.d.ts [hunk 1]
```diff
@@ -64,4 +64,15 @@ declare module 'libsession_util_nodejs' {
...
   export type BaseWrapperActionsCalls = MakeWrapperActionCalls<BaseConfigWrapper>;
+
+  export type ConstantsType = {
+    /** 100 bytes */
+    CONTACT_MAX_NAME_LENGTH: number;
+    /** 100 bytes - for legacy groups and communities */
+    BASE_GROUP_MAX_NAME_LENGTH: number;
+    /** 100 bytes */
+    GROUP_INFO_MAX_NAME_LENGTH: number;
+  }
```
Comment (Bilb) [inline shared.d.ts]: nice to have the jsdoc, but also easy to get out of sync if ever libsession changes it

**Comment by Bilb** [comment]
lgtm

File: shared.d.ts (hunk 1 above)
Comment (yougotwill) [reply shared.d.ts]: Yeah not an ideal solution if we can find a way to link or reference the C++ directly then I'm all for it.

File: package.json [hunk 2]
```diff
@@ -9,14 +9,16 @@
     "email": "team@oxen.io"
   },
   "scripts": {
+    "clean": "rimraf .cache && rimraf build",
```
Comment (Bilb) [inline package.json]: you can just have 
`rimraf .cache build`
that will avoid keeping the build, when cache is empty

**Review by Bilb — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (yougotwill) [reply package.json]: Fixed!

**Review by Bilb — APPROVED** [review]

**Pull request merged**